src/ui/public/agg_types/buckets/histogram.js

```diff
@@ -27,12 +28,66 @@ export function AggTypesBucketsHistogramProvider(Private) {
       {
         name: 'interval',
         editor: intervalTemplate,
+        onSearchRequestStart(aggConfig, searchSource) {
+          const field = aggConfig.getField();
+          const aggBody = field.scripted
+            ? { script: { inline: field.script, lang: field.lang } }
+            : { field: field.name };
+
+          return searchSource
+            .extend()
+            .size(0)
+            .aggs({
+              maxAgg: {
+                max: aggBody
+              },
+              minAgg: {
+                min: aggBody
+              }
+            })
+            .fetchAsRejectablePromise()
+            .then((resp) => {
+              aggConfig.params.min = _.get(resp, 'aggregations.minAgg.value', null);
+              aggConfig.params.max = _.get(resp, 'aggregations.maxAgg.value', null);
+            });
+        },
         write: function (aggConfig, output) {
-          output.params.interval = parseFloat(aggConfig.params.interval);
+          let interval = parseFloat(aggConfig.params.interval);
+
+          // ensure interval does not create too many buckets and crash browser
+          if (aggConfig.params.min !== null && aggConfig.params.max != null) {
+            const range = aggConfig.params.max - aggConfig.params.min;
+            const bars = range / interval;
+            if (bars > config.get('histogram:maxBars')) {
+              const minInterval = range / config.get('histogram:maxBars');
+              // Round interval by order of magnitude to provide clean intervals
+              // Always round interval up so there will always be less buckets than histogram:maxBars
+              const orderOfMaginute = Math.pow(10, Math.floor(Math.log10(minInterval)));
+              let roundInterval = orderOfMaginute;
+              while (roundInterval < minInterval) {
+                roundInterval += orderOfMaginute;
+              }
+              interval = roundInterval;
+            }
+          }
+
+          output.params.interval = interval;
         }
       },
 
       {
+        name: 'max',
```

Comment:
We should probably use a less generic param name, maybe combine `min`/`max` to `autoBounds: { min/max }`?